### ISSUE TITLE
Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer

### ISSUE THREAD

**Issue by shnakazawa** [TeamAidemy/ds-paper-summaries#3]
Raffel, Colin, Noam Shazeer, Adam Roberts, Katherine Lee, Sharan Narang, Michael Matena, Yanqi Zhou, Wei Li, and Peter J. Liu. 2019. “Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer.” arXiv [cs.LG]. arXiv. http://arxiv.org/abs/1910.10683.

- 2017年にGoogleのチームが発表した仕事。
- Text-to-Text Transfer Transformer, 通称 T5. 
- 学習の入力から出力まで、全てをテキストで扱う (Tex-to-Text) フレームワークを導入。
- 全てテキスト = タスクごとにモデルの形を変える必要がない = 1個のモデルであらゆるタスクに対応できる！
- 2019年までに報告されていた有名どころの手法を比較検討。Transformerの総説論文的に読むのも面白い。
	- 2020年にリバイズで議論等追加。

ちゃんと他者の論文も引用し、残された課題なども議論している。好き。

## Abstract

> Transfer learning, where a model is first pre-trained on a data-rich task before being fine-tuned on a downstream task, has　 emerged as a powerful technique in natural language processing (NLP). The effectiveness of transfer learning has given rise to a diversity of approaches, methodology, and practice. In this paper, we explore the landscape of transfer learning techniques for NLP by introducing a unified framework that converts all text-based language problems into a text-to-text format. Our systematic study compares pre-training objectives, architectures, unlabeled data sets, transfer approaches, and other factors on dozens of language understanding tasks. By combining the insights from our exploration with scale and our new ``Colossal Clean Crawled Corpus'', we achieve state-of-the-art results on many benchmarks covering summarization, question answering, text classification, and more. To facilitate future work on transfer learning for NLP, we release our data set, pre-trained models, and code.

(DeepL翻訳)

自然言語処理(NLP)において、モデルを事前に学習させ、その後、下流のタスクで微調整を行う「転移学習」は、強力な手法として注目されている。転移学習の有効性は、様々なアプローチ、方法論、実践を生み出してきた。本論文では、全てのテキストベースの言語問題をテキストからテキストに変換する統一的なフレームワークを導入することで、NLPのための転移学習技術の展望を探っている。本論文では、数十の言語理解タスクについて、事前学習の目的、アーキテクチャ、ラベル無しデータセット、転移アプローチ、その他の要素を比較する。また、この研究で得られた知見を規模や新しい「Colossal Clean Crawled Corpus」と組み合わせることで、要約、質問応答、テキスト分類などをカバーする多くのベンチマークで最先端の結果を達成している。今後、自然言語処理における転移学習の研究を促進するために、我々のデータセット、事前学習済みモデル、そしてコードを公開する。

## コード

https://github.com/google-research/text-to-text-transfer-transformer

## 解決した課題/先行研究との比較

- Pre-training→Fine-tuningの有用性が示され、広く使われるようになってきた頃。(Vaswani et al., 2017 Transformer; Devlin et al., 2018 BERT; Radford et al., 2018 GPT; etc...)
- ラベルなしの教師なし学習を利用してpre-trainingを行う手法が提案されたことから、インターネット上のラベルのないテキストデータが大量に使用できるようになり、転移学習はより力を増してきた。
- 一方、分野が急激に拡大することで、異なるアルゴリズムの比較や、新しいアルゴリズムの効果を画一的に調べることが困難になってきたという一面も。
- 本論文では、全てのNLP問題を「テキストからテキストへ (Text-to-text)」問題として扱う、つまり入力もテキスト・出力もテキストとして扱うフレームワークを構築し、全てのタスクに同じモデルを適用できるようにすることで、さまざまな課題に対するさまざまなモデルのパフォーマンスを画一的に比較できるようにする手法を提案している。

## 技術・手法のポイント

- BERT (Devlin et al., 2018) をベースにしている。
	- パラメータ数がかなり違う。定量に用いられたモデルでは11 billion.
- Text-to-text: モデルは全てのタスクをテキスト入力として受け取り、それを学習し、ターゲットテキストを生成する。様々なタスクで同じモデル、同じ損失関数、同じハイパーパラメータが使用できる。
[![Image from Gyazo](https://i.gyazo.com/005803e6bb1d72cc015d77bb6d5f4b3d.png)](https://gyazo.com/005803e6bb1d72cc015d77bb6d5f4b3d)
- このアイデアを用いて、有名どころのモデル・手法を比較。
	- Pre-trainingの仕方なども、様々な手法を検討。
- テキストのトークン化には Sentence Piece (Kudo and Richardson, 2018) を使用。
- モデルが実行すべきタスクを指定するために、元の入力シーケンスにタスク固有の接頭辞を付加してからモデルに与える。

## 評価指標

- 24中18のタスクでState-of-the-Art!

State-of-the-Art
- Abstractive summarization
	- CNN/DM (ROUGE-1-F, ROUGE-2-F, ROUGE-L-F)
- Sentence acceptability judgement
	- CoLA (Matthew's)
- Sentiment analysis (感情分析)
	- SST-2 (Accuracy)
- Paraphrasing/sentence similarit
	- STS-B (Pearson, Spearman) 
	- QQP (F1)
- Natural language inference
	- MNLI-m (Accuracy)
	- MNLI-mm (Accuracy)
	- RTE (Accuracy)
	- CB (F1, Accuracy)
- Coreference resolutio
	- WNLI (Accuracy)
	- WSC (Accuracy)
- Sentence completio
	- COPA (Accuracy)
- Word sense disambiguation
	- WiC (Accuracy)
- Question answering
	- SQuAD (EM, F1)
	- MultiRC (F1a, EM)
 - ReCoRD (F1, Accuracy)
	- BoolQ (Accuracy)

ほぼSOTA
- Paraphrasing/sentence similarit
	- MRPC (F1, Accuracy)
	- QQP (Accuracy)
- Natural language inference
	- QNLI (Accuracy)
- WMT EnDe Translation (SacreBLEU v1.3.0)
- WMT EnFr Translation (SacreBLEU v1.3.0)

あまり良くない
- WMT EnRo Translation (SacreBLEU v1.3.0) ... 英ルーマニア語翻訳

(24タスクとどうやって数えているのだろう…？GLUE AverageとSuperGLUE Averageそれぞれもカウントしている？)

シンプルなアプローチにも関わらず、タスク固有のアーキテクチャに匹敵するパフォーマンスを獲得！

## 残された課題・議論

- モデルが大きいほどパフォーマンスが向上する傾向が示された。
	- 転移学習の利点の一つは少リソースで優れたパフォーマンスを出せる点。それが今後活きなくなってしまう懸念。
	- 転移学習をより有効に使う方法の検討が進められている (Hinton et al., 2015; Sanh et al., 2019; Jiao et al., 2019; Lan et al., 2019; Shazeer et al., 2017)
- 事前学習の工夫の可能性
	- 実在するテキストと機械が作ったテキストの区別をさせる (Clark et al., 2020)
		- Masked language modelingによる事前学習ではなく、以下の方法で事前学習を行う
			- 事前学習用の文章の一部の単語を、小さなGeneratorネットワークを用いて、もっともらしい別単語に置き換える
			- 事前学習では、モデルに各単語が元の文章のままか、置き換えられた単語かを2値分類させるよう訓練
		- これにより、モデルがMASKトークン周辺の情報だけでなく、文章全体を注意深く読むことになるために、事前学習のクオリティが上がる
	- Pre-trainingと実タスクの「類似性」を定式化し、用いる学習データに優先度をつける
		- コンピュータビジョンの分野では、Pre-trainingに使用するデータをどのように選択するかで、実タスクに対するパフォーマンスが変わることも報告されている (Huh et al., 2016; Kornblith et al., 2018; He et al., 2018).
		- タスク間の関連性を考慮しての教師あり事前学習におけるタスク選択はGLUEベンチマークスコアを改善する (Phang et al., 2018)
- 事前学習に英語のみを用いての翻訳タスク回答はスコアが悪かった。
	- 「トレーニングに用いた言葉が何語か」という情報がまだ必要。
	- これを不要にする方法、つまり「言語に関係なくNLPタスクを実行できるモデル」を作れないだろうか。
		- このチームが次に狙っているのはここだろう。

## 重要な引用

- Transformerと言語モデル
	- Vaswani, Ashish, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan N. Gomez, Lukasz Kaiser, and Illia Polosukhin. 2017. “Attention Is All You Need.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1706.03762.
	- Devlin, Jacob, Ming-Wei Chang, Kenton Lee, and Kristina Toutanova. 2018. “BERT: Pre-Training of Deep Bidirectional Transformers for Language Understanding.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1810.04805.
	- Radford, Alec, Karthik Narasimhan, Tim Salimans, and Ilya Sutskever. 2018. “Improving Language Understanding by Generative Pre-Training.” https://www.cs.ubc.ca/~amuham01/LING530/papers/radford2018improving.pdf.
- テキストのトークン化に持ちいた "SentencePiece"
	- Kudo, Taku, and John Richardson. 2018. “SentencePiece: A Simple and Language Independent Subword Tokenizer and Detokenizer for Neural Text Processing.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1808.06226.
- 転移学習をより有効に使う方法
	- Clark, Kevin, Minh-Thang Luong, Quoc V. Le, and Christopher D. Manning. 2020. “ELECTRA: Pre-Training Text Encoders as Discriminators Rather Than Generators.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/2003.10555.
	- Hinton, Geoffrey, Oriol Vinyals, and Jeff Dean. 2015. “Distilling the Knowledge in a Neural Network.” arXiv [stat.ML]. arXiv. http://arxiv.org/abs/1503.02531.
	- Sanh, Victor, Lysandre Debut, Julien Chaumond, and Thomas Wolf. 2019. “DistilBERT, a Distilled Version of BERT: Smaller, Faster, Cheaper and Lighter.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1910.01108.
	- Jiao, Xiaoqi, Yichun Yin, Lifeng Shang, Xin Jiang, Xiao Chen, Linlin Li, Fang Wang, and Qun Liu. 2019. “TinyBERT: Distilling BERT for Natural Language Understanding.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1909.10351.
	- Lan, Zhenzhong, Mingda Chen, Sebastian Goodman, Kevin Gimpel, Piyush Sharma, and Radu Soricut. 2019. “ALBERT: A Lite BERT for Self-Supervised Learning of Language Representations.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1909.11942.
	- Shazeer, Noam, Azalia Mirhoseini, Krzysztof Maziarz, Andy Davis, Quoc Le, Geoffrey Hinton, and Jeff Dean. 2017. “Outrageously Large Neural Networks: The Sparsely-Gated Mixture-of-Experts Layer.” arXiv [cs.LG]. arXiv. http://arxiv.org/abs/1701.06538.
	- Phang, Jason, Thibault Févry, and Samuel R. Bowman. 2018. “Sentence Encoders on STILTs: Supplementary Training on Intermediate Labeled-Data Tasks.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1811.01088.
